index.js

```diff
@@ -58,13 +58,45 @@ module.exports = (src, dest, opts) => {
 	return promise;
 };
 
-module.exports.sync = function (src, dest, opts) {
-	if (!src || !dest) {
-		throw new CpFileError('`src` and `dest` required');
+const checkSrcIsFile = function (src) {
+	const stat = fs.statSync(src);
+	if (stat.isDirectory()) {
+		throw Object.assign(new CpFileError(`EISDIR: illegal operation on a directory '${src}'`), {
```

Comment:
Isn't this handled by `copyFileSync` in `fs.js`? If Node.js throws an error on directories when using `fs.copyFileSync` we will re-throw it automatically in our wrapper.

Reply:
I think node expects both src and dest to exist, so if we skip it here, we'll create dest on the next line and THEN node may throw. Not super happy about it either

Reply:
It looks like `copyFile` handles directories too:

```js
'use strict';
const fs = require('fs');
fs.copyFileSync('foo', 'bar');
```

<img width="224" alt="screen shot 2017-09-14 at 14 04 50" src="https://user-images.githubusercontent.com/709159/30428926-c0336b82-9955-11e7-85d0-d5587a09e457.png">

Reply:
Ok, lemme try